### PR TITLE
[#1607] feat(jdbc): Filtering out the display of jdbc system databases.

### PR DESCRIPTION
**Pull request by Clearvive**
### What changes were proposed in this pull request?

Filtering out the display of jdbc system databases.

### Why are the changes needed?

Fix: #1607 

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
UT/IT

### CODE REVIEW

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java [hunk 1]
```diff
@@ -13,13 +13,28 @@
...
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Set;
 import org.apache.commons.collections4.MapUtils;
 import org.apache.commons.lang3.StringUtils;
 
 /** Database operations for MySQL. */
 public class MysqlDatabaseOperations extends JdbcDatabaseOperations {
+
+  public static final Set<String> SYS_MYSQL_DATABASE_NAMES =
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java:27]: Is there any other system databases in other versions? 
Is the database name case-senstive ,like "SYS" is a system database?
same question for PG.

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java [hunk 2]
```diff
@@ -13,13 +13,28 @@
...
+      Collections.unmodifiableSet(
+          new HashSet<String>() {
```
Comment (yuqi1129) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java:29]: I would suggest introduce immutable lib and use `ImmutableSet.of(...)` directly, It's quite tedious here.

**Comment by FANNG1** [comment]
LGTM, except the minor comments

**Comment by jerryshao** [comment]
Please fix the left comments @Clearvive thanks.

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlSchemaOperations.java [hunk 3]
```diff
@@ -114,4 +129,9 @@ protected Connection getConnection() throws SQLException {
     connection.setCatalog(database);
     return connection;
   }
+
+  @Override
+  protected boolean isSystemDatabase(String dbName) {
+    return SYS_PG_DATABASE_NAMES.contains(dbName);
```
Comment (jerryshao) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/operation/PostgreSqlSchemaOperations.java:135]: Could you change to like `SYS_PG_DATABASE_NAMES.contains(dbName.toLowCase...)`?

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java [hunk 4]
```diff
@@ -107,4 +122,9 @@ public JdbcSchema load(String databaseName) throws NoSuchSchemaException {
       throw this.exceptionMapper.toGravitinoException(se);
     }
   }
+
+  @Override
+  protected boolean isSystemDatabase(String dbName) {
+    return SYS_MYSQL_DATABASE_NAMES.contains(dbName.toLowerCase());
```
Comment (jerryshao) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java:128]: You'd better changing to `dbName.toLowerCase(Locale.ROOT)`.

**Review by jerryshao — APPROVED** [review]

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/TestMultipleJdbcLoad.java [hunk 5]
```diff
@@ -122,7 +122,7 @@ public void testCreateMultipleJdbc() throws URISyntaxException {
         postgreSqlCatalog
             .asSchemas()
             .listSchemas(Namespace.of(metalakeName, postgreSqlCatalogName));
-    Assertions.assertNotEquals(0, nameIdentifiers.length);
+    Assertions.assertEquals(0, nameIdentifiers.length);
```
Comment (FANNG1) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/TestMultipleJdbcLoad.java:125]: why check pg listSchemas returns empty while keeps  MySQL listSchema result not empty
```java

    NameIdentifier[] nameIdentifiers =
        mysqlCatalog.asSchemas().listSchemas(Namespace.of(metalakeName, mysqlCatalogName));
    Assertions.assertNotEquals(0, nameIdentifiers.length);
    nameIdentifiers =
        postgreSqlCatalog
            .asSchemas()
            .listSchemas(Namespace.of(metalakeName, postgreSqlCatalogName));
    Assertions.assertEquals(0, nameIdentifiers.length);
```

**Pull request merged**